src/Components/ProductFilters.jsx

```diff
@@ -41,16 +79,63 @@ export default function ProductFilters({ filterOptions, setFilterOptions, sortOp
               <legend className="block font-medium">Price</legend>
               <div className="pt-6 space-y-6 sm:pt-4 sm:space-y-4">
                 {filterOptions.price.map((option, optionIdx) => (
-                  <div key={option.minValue} className="flex items-center text-base sm:text-sm">
+                  <div
+                    key={option.minValue}
+                    className="flex items-center text-base sm:text-sm"
+                  >
                     <input
                       id={`price-${optionIdx}`}
                       name="price[]"
                       defaultValue={option.minValue}
                       type="checkbox"
                       className="flex-shrink-0 h-4 w-4 border-gray-300 rounded text-black focus:ring-black"
                       defaultChecked={option.checked}
+                      onClick={(e) => {
+                        filterOptions.price.forEach((element) => {
+                          if (element.minValue.toString() === e.target.value) {
+                            if (element.checked === false) {
+                              element.checked = true;
+                            } else if (element.checked === true)
+                              element.checked = false;
+                          }
+                        });
+                        let selected_array = [];
+
+                        let selected_products = filterOptions.price.filter(
+                          function (x) {
+                            return x.checked === true;
+                          }
+                        );
+
+                        selected_products.forEach((element) => {
+                          selected_array.push(
+                            allproducts.filter(function (item) {
+                              return (
+                                item.price > element.minValue &&
+                                item.price < element.maxValue
+                              );
+                            })
+                          );
+                        });
+                        if (selected_array.length > 0) {
+                          setProducts(selected_array.flat(1));
+                          setselectedByPrice(selected_array.flat(1));
+                        } else {
+                          if (selectedByColor.length > 0) {
+                            setProducts(selectedByColor);
+                          } else {
+                            setProducts(allproducts);
+                            setselectedByPrice(allproducts);
+                          }
+                        }
+
+                        set_Nbr_Filters_price(selected_array.length);
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** We should try and refactor this. It's quite a large chunk of code which is hard to maneuver at first glance. 

I also notice its very similar to some of the code below (160 - 206). Given it's size right, it helps to isolate this to a single helper function, and then call it from the onClick handlers.‏